remote/mattermost/remote.go

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package mattermost
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/target/flottbot/models"
+	"github.com/target/flottbot/remote"
+)
+
+// Client struct.
+type Client struct {
+	Server string
+	Token  string
+	BotID  string
+}
+
+type mmLogger struct {
+	zerolog.Logger
+}
+
+func (l *mmLogger) Output(_ int, s string) error {
+	l.Logger.Info().Msg(s)
+	return nil
+}
+
+// validate that Client adheres to remote interface.
+var _ remote.Remote = (*Client)(nil)
+
+// instantiate a new mattermost client.
+func (c *Client) new() *model.Client4 {
+	log.Info().Msgf("%#v", c)
+	api := model.NewAPIv4Client("http://" + c.Server)
```

Comment:
presumably server is not typically on `http` but `https`? should the expectation be to provide a FQDN for the `MATTERMOST_SERVER` configuration item?

Reply:
Yea, it probably should default to `https`.

I'd like to keep the ability to set `MATTERMOST_SERVER` as a bare IP address, if for no other reason than testing.

What's the best way to allow the user to connect over `http` if that's what they want? An optional `MATTERMOST_SECURE_PROTOCOL` variable or something like that?

Reply:
i think i would default to secure: `https` + `wss` and add a flag like you said. maybe inverse though, something like `MATTERMOST_USE_INSECURE` or `MATTERMOST_LOCAL`? not sure if `local` is the only use-case, so maybe the former?